src/main/java/org/sonarqube/gradle/SonarQubePlugin.java

```diff
@@ -223,8 +223,12 @@ static void appendProp(Map<String, Object> properties, String key, Object valueT
   }
 
   static void setTestClasspathProps(Map<String, Object> properties, Collection<File> testClassDirs, Collection<File> testLibraries) {
-    appendProps(properties, "sonar.java.test.binaries", testClassDirs);
-    appendProps(properties, "sonar.java.test.libraries", testLibraries);
+    appendProps(properties, "sonar.java.test.binaries", exists(testClassDirs));
+    appendProps(properties, "sonar.java.test.libraries", exists(testLibraries));
+  }
+
+  private static List<File> exists(Collection<File> files) {
+    return files.stream().filter(File::exists).collect(Collectors.toList());
```

Comment:
If this method is only used with directories, perhaps it would be better to use `isDirectory`, which implies that it exists.

Reply:
For sonar.java.libraries, it can be files (.jar)